src/dataflow/src/server.rs

```diff
@@ -544,13 +546,18 @@ where
             }
         }
 
-        for (id, history) in self.render_state.ts_histories.iter() {
+        for (id, source_probes) in self.render_state.source_uppers.iter() {
             // Read the upper frontier and compare to what we've reported.
-            history.read_upper(&mut new_frontier);
             let prev_frontier = self
                 .reported_frontiers
                 .get_mut(&id)
                 .expect("Source frontier missing!");
+            let mut new_frontier = prev_frontier.clone();
```

Comment:
i think the biggest functional incompleteness here is that im not sure what to do if the source gets closed.

Reply:
You mean because we wouldn't report any more progress? Or because you don't know when to remove the frontier from `reported_frontiers`?

Related: the initial frontier is inserted into `reported_frontiers` when getting an `AddSourceTimestamping`. That feels wrong and we should maybe insert a frontier the first time we render a source of that `id`. But then I don't know when to remove that frontier again. It seems we need some kind of refcounting, where we remove it when the last rendered source is dropped?

Reply:
We have that count already, btw: the `len()` of the `Vec` in `source_uppers`.

Reply:
sorry i meant "if a source instance is dropped" - so when we drop a source instance we'll want to delete this probe handle as well (because we don't want the frontier to spuriously advance to the empty frontier and we want the dropped dataflow's resources to be reclaimed) and I don't know a great way to do that.

I can think of some solutions but (e.g. we keep a map from source id -> (source_instance_id, ProbeHandle) but those seem like exactly the kind of complexity increasing things I wanted to avoid.

Reply:
potentially complicated, but: the source handles we keep in there could be weak references, and the real reference is kept in the source token, that eventually gets dropped when the source instance is dropped.

I don't know how much I like it, though. But I think the source token dropping is the only point where we notice a source being dropped?

Reply:
done - mostly copying the approach used in `ts_source_mapping`. This + a bugfix to use the proper source id instead of the transient "bare source" id fixes almost all the bugs in this pr.

Reply:
The behavior elsewhere is to advance all of the frontiers to the empty set, I believe. This happens naturally as part of dataflow shutdown, so nothing too much to do if you want that behavior. I *think* this lines up with "this is COMPUTE feedback", and certainly shouldn't be taken to mean "this source has been drained dry". I understand that this is what it is currently hooked up, though!

Reply:
We chatted on slack but to summarize:

1. the probe has been moved to the boundary between storage and compute, and is now meant to reflect the per-source upper, rather than the per-source-instance upper. If a source instance is dropped, it will never be consulted afterwards to determine the source's upper, but if a source instance drops its capability and stays on (file sources without tail seem to do this) that source will present an upper == the empty antichain (verified with td test).

2. We should break apart some commands into `StorageCommand` and `ComputeCommand` e.g. `AllowCompaction` for sources (owned by storage) vs for views (owned by compute).

I don't feel like we need to do that in this PR specifically, but am happy to prototype a followup in a separate pr if we want to see what it looks like before merging this one?